phabricatoremails/error_notify.py

```diff
@@ -0,0 +1,24 @@
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import traceback
+from dataclasses import dataclass
+from typing import Any
+
+import sentry_sdk
+from statsd import StatsClient
+
+
+@dataclass
+class ErrorNotify:
+    _logger: Any
+    _stats: StatsClient
+
+    def notify(self, exception: Exception, warning: str, failure_stat: str):
+        formatted_exception = traceback.format_exception(
+            type(exception), exception, exception.__traceback__
+        )
+        self._logger.warning(formatted_exception)
```

Comment:
What is the benefit of formatting an exception with `traceback.format_exception` and then logging it as a warning, rather than logging it as an exception (which retains the stack trace in the logs)?

Reply:
Good question, this one I was on the fence on. `logger.exception(...)` only works when executed from an error handler. However, that adds an implicit calling requirement on `ErrorNotify.notify(...)`.
Instead, I figured that by only requiring the exception (then always working, regardless of context), this method wouldn't have any unexpected surprises in the future.

Reply:
I think a middle of the road solution would be to log the exception where it happens, and use this notifier for sentry and stats, since we're not using the formatted exception anywhere else. I'd rather not change the log level to warning just to work around the intended use of `logger.exception`.